src/Sylius/Bundle/ApiBundle/Provider/CaptureUrlPaymentConfigurationProvider.php

```diff
@@ -0,0 +1,79 @@
+<?php
```

Comment:
Instead of decorating the composite provider, why not implement its subprovider?

```php
<?php

/*
 * This file is part of the Sylius package.
 *
 * (c) Paweł Jędrzejewski
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */

declare(strict_types=1);

namespace Sylius\Bundle\ApiBundle\Provider;

use Payum\Core\Payum;
use Payum\Core\Security\TokenInterface;
use Sylius\Bundle\ApiBundle\Payment\PaymentConfigurationProviderInterface;
use Sylius\Bundle\PayumBundle\Model\GatewayConfigInterface;
use Sylius\Component\Core\Model\PaymentInterface;
use Sylius\Component\Core\Model\PaymentMethodInterface;
use Webmozart\Assert\Assert;

final class CaptureUrlPaymentConfigurationProvider implements PaymentConfigurationProviderInterface
{
    public function __construct(
        private readonly Payum $payum
    ) {
    }

    public function supports(PaymentMethodInterface $paymentMethod): bool
    {
        return $paymentMethod->getGatewayConfig() instanceof GatewayConfigInterface;
    }

    public function provideConfiguration(PaymentInterface $payment): array
    {
        $token = $this->provideTokenBasedOnPayment($payment);

        return [
            'capture_url' => $token->getTargetUrl(),
        ];
    }

    private function provideTokenBasedOnPayment(PaymentInterface $payment): TokenInterface
    {
        $gatewayConfig = $this->getGatewayConfig($payment);
        $config = $gatewayConfig->getConfig();
        $gatewayName = $gatewayConfig->getGatewayName();
        $tokenFactory = $this->payum->getTokenFactory();

        $afterToken = $tokenFactory->createToken(
            $gatewayName,
            $payment,
            $config['sylius_api_after_path'] ?? 'api_payums_shop_after_pay_item',
            $config['sylius_api_after_path_parameters'] ?? [],
        );

        return $tokenFactory->createToken(
            $gatewayName,
            $payment,
            'api_payums_shop_capture_item',
            [],
            $afterToken->getTargetUrl()
        );
    }

    private function getGatewayConfig(PaymentInterface $payment): GatewayConfigInterface
    {
        /** @var PaymentMethodInterface|null $paymentMethod */
        $paymentMethod = $payment->getMethod();
        Assert::notNull($paymentMethod);

        /** @var GatewayConfigInterface|null $gatewayConfig */
        $gatewayConfig = $paymentMethod->getGatewayConfig();
        Assert::notNull($gatewayConfig);

        return $gatewayConfig;
    }
}
```

Then you just need to tag it with `sylius.api.payment_method_handler`

Reply:
BTW:
1. I'm not sure why PaymentConfigurationProviderInterface is in its own namespace next to `Provider`, but perhaps it would make sense to move it next to other providers (alternatively, my version of this provider could be moved into the `Payment` namespace).
2. There's no automatic tagging of the instances of this interface with `sylius.api.payment_method_handler`, but, assuming Sylius tags instances of other interfaces where appropriate, doing so here seems appropriate as well.
3. The composite provider does not mention PaymentConfigurationProviderInterface interface anywhere. It would make sense to do so, at least to improve discoverability.